extensions/ql-vscode/src/databases.ts

```diff
@@ -430,21 +430,23 @@ class DatabaseItemImpl implements DatabaseItem {
  */
 function eventFired<T>(event: vscode.Event<T>, timeoutMs = 1000): Promise<T | undefined> {
   return new Promise((res, _rej) => {
-    let timeout: NodeJS.Timeout | undefined;
-    let disposable: vscode.Disposable | undefined;
-    function dispose() {
-      if (timeout !== undefined) clearTimeout(timeout);
-      if (disposable !== undefined) disposable.dispose();
-    }
-    disposable = event(e => {
-      res(e);
-      dispose();
-    });
-    timeout = setTimeout(() => {
+    const timeout = setTimeout(() => {
       logger.log(`Waiting for event ${event} timed out after ${timeoutMs}ms`);
       res(undefined);
       dispose();
     }, timeoutMs);
+    const disposable = event(e => {
+      res(e);
+      dispose();
+    });
+    function dispose() {
+      if (timeout !== undefined) {
```

Comment:
Huh, somehow I didn't think typescript let me forward-reference this function recursively. If it does, these checks for `!== undefined` aren't necessary anymore, are they?

Reply:
`timeout !== undefined` is never needed since `clearTimeout` is a noop for non-timers. And it looks like the other if statement is not necessary either.